unknown type error during compilation

when trying to compile a program using `libsignify` i am getting an error about an unknown type (included below). what type should this be in the code? i was just trying to replicate the `verify()` from `signify-rs` in a small program.

rustc version:
```
> rustc --version
rustc 1.60.0 (7737e0b5c 2022-04-04)
```

error:
```
error[E0282]: type annotations needed
   --> D:\<MYUSER>\.cargo\registry\src\github.com-1ecc6299db9ec823\libsignify-0.5.2\src\key.rs:176:42
    |
176 |         checksum.copy_from_slice(&digest.as_ref()[0..8]);
    |                                   -------^^^^^^--
    |                                   |      |
    |                                   |      cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                                   this method call resolves to `&T`
    |
    = note: type must be known at this point

For more information about this error, try `rustc --explain E0282`.
```

--- a/libsignify/src/key.rs
+++ b/libsignify/src/key.rs
@@ -173,7 +173,7 @@ impl PrivateKey {
     fn calculate_checksum(complete_key: &UnencryptedKey) -> [u8; 8] {
         let digest = Sha512::digest(complete_key.0.as_ref());
         let mut checksum = [0; 8];
-        checksum.copy_from_slice(&digest.as_ref()[0..8]);
+        checksum.copy_from_slice(&digest.as_slice()[0..8]);
         checksum
     }
 

